lib/WebLedgerClient.js

```diff
@@ -61,36 +61,41 @@ class WebLedgerClient {
 
     const ledgerAgent = await this.getAgent();
     const baseURL = ledgerAgent.service.ledgerQueryService;
-    const queryServiceApi =
-      create({baseURL, httpsAgent: this.httpsAgent, timeout: this.timeout});
 
     this.logger.log(`Retrieving Record [ledger: ${this.hostname}]:`, id);
-
-    const response = await queryServiceApi.post(
-      `?id=${encodeURIComponent(id)}`, {}, {
-        headers: constants.DEFAULT_HEADERS
+    let body = {};
```

Comment:
```suggestion
    let body;
```

Reply:
I kind of disagree on this one. If there is no body then the check for record results in `could not read property record or undefined`. While initializing body to an empty record means the error is: `Fetched Record has empty body/"record" property.', 'DataError'`

The later strikes me as more informative, the former as merely a fail case no one thought of.

https://github.com/digitalbazaar/web-ledger-client/blob/100547852e0e38e462d30ef97586e51c6e547ddd/lib/WebLedgerClient.js#L103-L109

Reply:
I think Mandy's suggestion on L66 is good.  But the real issue is on L76.  If `response.data` is undefined etc, an error should at that point with some useful error message.  I think it's even the case that we should move the test for `response.data.record` into this section as well.  In other words, refactor the error handling to include:

If we don't get a `record` back, we should just throw a good error and be done. 

https://github.com/digitalbazaar/web-ledger-client/blob/8df0321dc6bd90d9783558c545c4e191e45eebde/lib/WebLedgerClient.js#L110-L114